packages/gatsby-theme-docs/README.md

```diff
@@ -70,6 +70,8 @@ The project structure should contain at least the following files and folders:
 
   - `beta` (_optional_): indicates that the website should be marked as **beta**. Each page gets a beta flag, no matter if the page frontmatter has it defined or not. Furthermore, in the main navigation, the beta flag is shown near the website title and not next to each link.
 
+  - `hasReleaseNotes` (_optional_): indicates that the website should include links to the Release Notes pages, in the main navigation.
```

Comment:
I see the point but when you start developing you can just use `process.env.NODE_ENV` to determine if the release notes should be generated on production or not.

For the noindex we already have the flag.

As for the gatsby node setup, we are leveraging gatsby schema inference for e.g. the frontmatter fields, which works one you have "real" pages. Not having pages would mean that we need to patch the schema types to include those "optional" page fields.
So I would prefer to only enable the release notes setup based on the option flag.

Reply:
Let's speak about that, Im not sure we are speaking about the same thing? 

We can choose to start without an explicit schema now but my experience in the API Types and API Endpoints implementation is that you start to handle so many special cases and user errors that you end up with a fixed schema in any case.  
Here we would have to define the Schema for the `MdxFields` type and I can imaging that that would make a lot of the code simpler - maybe don't even have to have a config flag at all?  

Another advantage of explicitly defined schemata is that the error message that a user gets when they put schema-incompatible data in a document's frontmatter. With a defined schema the error is describing the actual problem.  With schema inference it's very hard to understand the error message because it just says that you have incompatible data on one document - and it's even random depending on which document is read first, it can happen that the correct document is complained about because the wrong one was read first. 

TL;DR:  Inferred schema can get really annoying to users.  If it's possible to define the `MdxFields` schema I vote for getting rid of the config flag altogether until we know more about the use cases.   Otherwise let's keep it as and do a refactor when the case comes up.  

PS: noindex is for the whole site but the situation I am speaking about is a site that is already indexed with content but gets release notes later.  Let's address that in a separate change when somebody really needs it. 

PPS: "Just" modifying code to  look into NODE_ENV isn't really an option for authors, but also here - we can address that if a case really comes up .

Reply:
For the content pages we are also not using a self-defined schema. If that's the direction you want to go, it's fine by me.

Let me have another look then if we can remove the `hasReleaseNotes` flags altogether and infer that from the query results.

> noindex is for the whole site but the situation I am speaking about is a site that is already indexed with content but gets release notes later. Let's address that in a separate change when somebody really needs it.

The noindex option is available both globally but also on a per-page level.

Reply:
Alright, I changed things a bit and created our own types and nodes. This simplifies several things actually, so it was a good thing to look into it. @nkuehn Thanks for pointing that out!

See https://github.com/commercetools/commercetools-docs-kit/pull/336/commits/5ba7b737a182236babf575b9dd1ab9397cbaac44

PS: I followed the example described in this article: https://www.christopherbiscardi.com/post/constructing-query-types-in-themes